convertor: panic on BuildKit attestation manifests

### What happened in your environment?

The userspace convertor panics when an OCI index contains a BuildKit attestation manifest, such as an image built with `docker buildx --provenance=true`, `--sbom=true`, or `--attest`. Provenance is enabled by default with the Buildx `docker-container` driver since v0.10.

`graphBuilder.process()` dispatches index entries by `Descriptor.MediaType`. An attestation manifest has the same media type as a runnable image manifest (`application/vnd.oci.image.manifest.v1+json`), so the convertor passes it to `buildOne()`.

BuildKit identifies these entries with an `unknown/unknown` platform and the annotation `vnd.docker.reference.type: attestation-manifest`. The manifest has an empty `config.RootFS.DiffIDs` list but one or more `application/vnd.in-toto+json` layers. `NewOverlayBDBuilderEngine()` then indexes the empty DiffID list while iterating those layers:

```
panic: index out of range [0] with length 0
```

`NewTurboOCIBuilderEngine()` has the same assumption. The convertor does not inspect the attestation annotation or platform, and it has no platform flag to exclude the entry.

Related changes are #266, which added multi-platform index traversal, and #180, which handled index layers but not attestations.

### What did you expect to happen?

Convert runnable platform manifests without treating in-toto attestations as rootfs layers. Preferably, copy attestation manifests through unchanged so provenance and SBOM data remain attached to the output index.

The open question is whether cross-repository conversion should copy the referenced attestation blobs or omit the descriptor with a warning.

### How can we reproduce it?

```bash
docker buildx create --use --driver docker-container
docker buildx build --provenance=true -t <reg>/attest-demo:latest --push .
docker buildx imagetools inspect <reg>/attest-demo:latest --raw

convertor \
  --repository <reg>/attest-demo \
  --input-tag latest \
  --overlaybd latest-obd \
  --oci
```

The raw index contains an `unknown/unknown` attestation manifest. Conversion panics in `NewOverlayBDBuilderEngine()`.

BuildKit documentation:

- https://docs.docker.com/build/metadata/attestations/attestation-storage/
- https://docs.docker.com/build/metadata/attestations/slsa-provenance/

### What is the version of your Accelerated Container Image?

`main` at `7d7bb97` (`containerd/v2 v2.0.9`, `image-spec v1.1.0`).

### What is your OS environment?

Linux. The failure is in manifest processing and does not appear OS-specific.

### Are you willing to submit PRs to fix it?

- [x] Yes, I am willing to fix it.

--- a/cmd/convertor/builder/builder.go
+++ b/cmd/convertor/builder/builder.go
@@ -91,6 +91,20 @@ type graphBuilder struct {
 	id        atomic.Int32
 }
 
+const (
+	attestationManifestAnnotation = "vnd.docker.reference.type"
+	attestationManifestType       = "attestation-manifest"
+)
+
+// isAttestationManifest reports whether desc is a BuildKit attestation manifest
+// (provenance/SBOM). Detection uses only the explicit
+// vnd.docker.reference.type=attestation-manifest annotation: the accompanying
+// unknown/unknown platform is a BuildKit convention shared by other artifacts,
+// so it is not a reliable identifier on its own.
+func isAttestationManifest(desc v1.Descriptor) bool {
+	return desc.Annotations[attestationManifestAnnotation] == attestationManifestType
+}
+
 func (b *graphBuilder) Build(ctx context.Context) error {
 	fetcher, err := b.Resolver.Fetcher(ctx, b.Ref)
 	if err != nil {
@@ -150,6 +164,17 @@ func (b *graphBuilder) process(ctx context.Context, src v1.Descriptor, tag bool)
 		for _i, _m := range index.Manifests {
 			i := _i
 			m := _m
+			if isAttestationManifest(m) {
+				// BuildKit attestation manifests (provenance/SBOM) carry no
+				// rootfs layers and reference a source platform manifest by
+				// digest (vnd.docker.reference.digest). That digest changes on
+				// conversion, so the attestation cannot be re-associated and
+				// converting its zero-layer manifest panics. Drop it from the
+				// converted index (with a warning) rather than emitting a stale
+				// reference.
+				log.G(ctx).Warnf("skipping attestation manifest %s: cannot be re-associated after conversion", m.Digest)
+				continue
+			}
 			wg.Add(1)
 			b.group.Go(func() error {
 				defer wg.Done()
@@ -165,6 +190,16 @@ func (b *graphBuilder) process(ctx context.Context, src v1.Descriptor, tag bool)
 		if ctx.Err() != nil {
 			return v1.Descriptor{}, ctx.Err()
 		}
+
+		// Drop attestation manifests from the converted index: their
+		// source-digest references no longer point at an image in the index.
+		filtered := index.Manifests[:0]
+		for _, m := range index.Manifests {
+			if !isAttestationManifest(m) {
+				filtered = append(filtered, m)
+			}
+		}
+		index.Manifests = filtered
 
 		// upload index
 		if b.Referrer {

--- a/cmd/convertor/builder/builder_attestation_test.go
+++ b/cmd/convertor/builder/builder_attestation_test.go
@@ -1,0 +1,78 @@
+/*
+   Copyright The Accelerated Container Image Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package builder
+
+import (
+	"testing"
+
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+func TestIsAttestationManifest(t *testing.T) {
+	tests := []struct {
+		name string
+		desc v1.Descriptor
+		want bool
+	}{
+		{
+			name: "attestation annotation",
+			desc: v1.Descriptor{
+				Annotations: map[string]string{
+					"vnd.docker.reference.type": "attestation-manifest",
+				},
+			},
+			want: true,
+		},
+		{
+			name: "attestation annotation with unknown platform",
+			desc: v1.Descriptor{
+				Annotations: map[string]string{
+					"vnd.docker.reference.type": "attestation-manifest",
+				},
+				Platform: &v1.Platform{OS: "unknown", Architecture: "unknown"},
+			},
+			want: true,
+		},
+		{
+			name: "unknown platform without attestation annotation is not an attestation",
+			desc: v1.Descriptor{
+				Platform: &v1.Platform{OS: "unknown", Architecture: "unknown"},
+			},
+			want: false,
+		},
+		{
+			name: "normal image manifest",
+			desc: v1.Descriptor{
+				Platform: &v1.Platform{OS: "linux", Architecture: "amd64"},
+			},
+			want: false,
+		},
+		{
+			name: "descriptor without attestation metadata",
+			desc: v1.Descriptor{},
+			want: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := isAttestationManifest(tt.desc); got != tt.want {
+				t.Errorf("isAttestationManifest() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}

